pkg/log/checkpoint/files_test.go

```diff
@@ -0,0 +1,128 @@
+package checkpoint
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"runtime"
+	"sort"
+	"testing"
+
+	"github.com/kolide/kit/ulid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFilesFound(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
+
+	var tests = []struct {
+		dirsToCreate int
+		filesPerDir  int
+	}{
+		{dirsToCreate: 2, filesPerDir: 2},
+		{dirsToCreate: 0, filesPerDir: 0},
+	}
+
+	for _, tt := range tests {
+		t.Run("testFilesFound", func(t *testing.T) {
+			dirs, expectedPaths, err := createTestFiles(tempDir, tt.dirsToCreate, tt.filesPerDir)
+			require.NoError(t, err, "creating test files")
+
+			foundPaths := fileNamesInDirs(dirs...)
+			sort.Strings(foundPaths)
+			require.Equal(t, expectedPaths, foundPaths)
+			require.Equal(t, tt.dirsToCreate*tt.filesPerDir, len(foundPaths))
+		})
+	}
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
+}
+
+func TestDirNotFound(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
+
+	const pathFmt = "%s/%s"
+	nonExistantDirs := []string{
+		fmt.Sprintf(pathFmt, tempDir, ulid.New()),
+		fmt.Sprintf(pathFmt, tempDir, ulid.New()),
+	}
+
+	expectedOutput := []string{}
+
+	for _, dir := range nonExistantDirs {
+		pathErr := fs.PathError{
+			Op:   "open",
+			Path: dir,
+			// would expect this to be constant or var somewhere in os package, but couldn't find
+			Err: errors.New("no such file or directory"),
+		}
+
+		// not found error is different for windows
+		if runtime.GOOS == "windows" {
+			pathErr.Err = errors.New("The system cannot find the file specified.")
+		}
+
+		expectedOutput = append(expectedOutput, pathErr.Error())
+	}
+
+	foundPaths := fileNamesInDirs(nonExistantDirs...)
+
+	require.Equal(t, expectedOutput, foundPaths)
+	require.Equal(t, len(nonExistantDirs), len(foundPaths))
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
+}
+
+func TestDirEmpty(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
+
+	dirs, _, err := createTestFiles(tempDir, 2, 0)
+	require.NoError(t, err, "creating test dirs")
+
+	expectedOutput := []string{}
+	for _, dir := range dirs {
+		expectedOutput = append(expectedOutput, emptyDirMsg(dir))
+	}
+
+	foundPaths := fileNamesInDirs(dirs...)
+
+	require.Equal(t, expectedOutput, foundPaths)
+	require.Equal(t, len(dirs), len(foundPaths))
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
+}
+
+func createTestFiles(baseDir string, dirCount int, filesPerDir int) (dirs []string, files []string, err error) {
+	files = []string{}
+
+	for i := 0; i < dirCount; i++ {
+		dir, err := os.MkdirTemp(baseDir, "notable-files-unit-test")
```

Comment:
If dirCount is > 1, won't they step on eachother?

Reply:
`os.MkdirTemp()` adds random string at the end of the pattern (second arg)

Reply:
I totally forgot. Thanks!